packages/compose/src/hooks/use-viewport-match/test/index.js

```diff
@@ -25,105 +25,104 @@ describe( 'useViewportMatch', () => {
 
-		expect( useMediaQueryMock.mock.calls ).toEqual( [
-			[ '(min-width: 1440px)' ],
-			[ '(max-width: 960px)' ],
-			[ '(max-width: 480px)' ],
-		] );
+		rerender( <TestComponent breakpoint="mobile" operator="<" /> );
 
-		root.unmount();
+		expect( container ).toHaveTextContent( 'useViewportMatch: false' );
+
+		expect( useMediaQueryMock ).toHaveBeenCalledTimes( 3 );
+		expect( useMediaQueryMock ).toHaveBeenNthCalledWith(
+			1,
+			'(min-width: 1440px)'
+		);
+		expect( useMediaQueryMock ).toHaveBeenNthCalledWith(
+			2,
+			'(max-width: 960px)'
+		);
+		expect( useMediaQueryMock ).toHaveBeenNthCalledWith(
+			3,
+			'(max-width: 480px)'
+		);
 	} );
 
 	it( 'should correctly simulate a value', async () => {
-		let root;
 		useMediaQueryMock.mockReturnValue( true );
 
 		const innerElement = <TestComponent breakpoint="wide" operator=">=" />;
 		const WidthProvider = useViewportMatch.__experimentalWidthProvider;
 
-		await act( async () => {
-			root = create(
-				<WidthProvider value={ 300 }>{ innerElement }</WidthProvider>
-			);
-		} );
-		expect( root.toJSON() ).toBe( 'useViewportMatch: false' );
-
-		await act( async () => {
-			root.update(
-				<WidthProvider value={ 1200 }>{ innerElement }</WidthProvider>
-			);
-		} );
-		expect( root.toJSON() ).toBe( 'useViewportMatch: false' );
-
-		await act( async () => {
-			root.update(
-				<WidthProvider value={ 1300 }>{ innerElement }</WidthProvider>
-			);
-		} );
-		expect( root.toJSON() ).toBe( 'useViewportMatch: true' );
-
-		await act( async () => {
-			root.update(
-				<WidthProvider value={ 1300 }>
-					<TestComponent breakpoint="wide" operator="<" />
-				</WidthProvider>
-			);
-		} );
-		expect( root.toJSON() ).toBe( 'useViewportMatch: false' );
-
-		expect( useMediaQueryMock.mock.calls ).toEqual( [
-			[ undefined ],
-			[ undefined ],
-			[ undefined ],
-			[ undefined ],
-		] );
-
-		root.unmount();
+		const { container, rerender } = render(
+			<WidthProvider value={ 300 }>{ innerElement }</WidthProvider>
+		);
+		expect( container ).toHaveTextContent( 'useViewportMatch: false' );
+
+		rerender(
+			<WidthProvider value={ 1200 }>{ innerElement }</WidthProvider>
+		);
+
+		expect( container ).toHaveTextContent( 'useViewportMatch: false' );
+
+		rerender(
+			<WidthProvider value={ 1300 }>{ innerElement }</WidthProvider>
+		);
+
+		expect( container ).toHaveTextContent( 'useViewportMatch: true' );
+
+		rerender(
+			<WidthProvider value={ 1300 }>
+				<TestComponent breakpoint="wide" operator="<" />
+			</WidthProvider>
+		);
+
+		expect( container ).toHaveTextContent( 'useViewportMatch: false' );
+
+		expect( useMediaQueryMock ).toHaveBeenCalledTimes( 4 );
+		expect( useMediaQueryMock ).toHaveBeenNthCalledWith( 1, undefined );
+		expect( useMediaQueryMock ).toHaveBeenNthCalledWith( 2, undefined );
+		expect( useMediaQueryMock ).toHaveBeenNthCalledWith( 3, undefined );
+		expect( useMediaQueryMock ).toHaveBeenNthCalledWith( 4, undefined );
```

Comment:
Suggestion (non-blocking): Maybe we should add an inline comment here that `useViewportMatch` receives `undefined` when the width is simulated. I had to check the source to see why the test was expecting undefined.

Reply:
Good point - added in 85a5d1d8064a732e21e627e536f9e4272179893d